### PR TITLE
:sectnumlevels: attribute documented

### PR DESCRIPTION
**Pull request by jaredmorgs**
Resolves Issue #383 by adding the :sectnumlevels: attribute to the documentation. Based the command line and API guidance on the :sectnums: attribute, I added similar override information as this seemed to be important in #288.

### CODE REVIEW

**Comment by mojavelinux** [comment]
> Based the command line and API guidance on the :sectnums: attribute, I added similar override information...

Actually, this is not true. Toggling this value midstream is not recognized. `sectnumlevels` is a "header only" attribute, which means it has no meaning outside of the document header.

(Btw, the only attribute that doesn't get "locked" when set via the API or CLI is `sectnums`. It's unique in this way. We call this a "flexible attribute" because it can always be changed midstream no matter how it is declared).

File: docs/_includes/sections.adoc [hunk 1]
```diff
@@ -170,13 +171,18 @@ If `sectnums` is set on the command line (or API), that overrides the value set
...
 ----
 include::ex-section.adoc[tags=sectnuml]
 ----
-<1> When the sectnumlevels attribute is assigned a value of 2, level 3 through level 5 section titles will not be prefixed with a number.
-////
+<1> When the `sectnumlevels` attribute is assigned a value of 2, level 3 through level 5 section titles are not prefixed with a number.
+
+If `sectnumlevels` is set on the command line (or API), it overrides the value set in the document header, but it does not prevent the document from toggling the value for regions of the document.
+
+If `sectnumlevels!` is set on the command line (or API), the numbers are disabled regardless of toggling within the document.
```
Comment (mojavelinux) [inline docs/_includes/sections.adoc]: These points about toggling the attribute value doesn't apply for `sectnumlevels`.

**Comment by jaredmorgs** [comment]
Ahh, good to know. I'll correct the info and push another PR. Are you happy
with the rest of the content?

On Mon, 18 Jan 2016 at 07:52 Dan Allen <notifications@github.com> wrote:

> Based the command line and API guidance on the :sectnums: attribute, I
> added similar override information...
>
> Actually, this is not true. Toggling this value midstream is not
> recognized. sectnumlevels is a "header only" attribute, which means it
> has no meaning outside of the document header.
>
> (Btw, the only attribute that doesn't get "locked" when set via the API or
> CLI is sectnums. It's unique in this way. We call this a "flexible
> attribute" because it can always be changed midstream no matter how it is
> declared).
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/asciidoctor/asciidoctor.org/pull/477#issuecomment-172386534>
> .
>
-- 

Sent from Mobile.

File: docs/_includes/sections.adoc [hunk 2]
```diff
@@ -170,13 +171,18 @@ If `sectnums` is set on the command line (or API), that overrides the value set
 
 If `sectnums!` is set on the command line (or API), then the numbers are disabled regardless of toggling within the document.
 
-////
-You can also restrict section numbering with the +sectnumlevels+ attribute.
+=== Numbering Depth
```
Comment (mojavelinux) [inline docs/_includes/sections.adoc]: We should use sentence capitalization for titles (though, admittedly, it's not consistent atm).

File: docs/_includes/sum-section.adoc [hunk 3]
```diff
@@ -34,4 +34,10 @@ Included in:
 |+:sectnums:+
 |Section numbering is off by default.
 Can be toggled on or off through document.
+
+|sectnumlevels
+|1-9
```
Comment (mojavelinux) [inline docs/_includes/sum-section.adoc]: The correct range is 0-5. (0 is effectively equivalent to disabling section numbering).

**Comment by mojavelinux** [comment]
I just finished the docs review.

**Comment by jaredmorgs** [comment]
Thanks for the feedback, Dan. Will make these changes and update the
original commit. :D

On Mon, 18 Jan 2016 at 07:58 Dan Allen <notifications@github.com> wrote:

> I just finished the docs review.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/asciidoctor/asciidoctor.org/pull/477#issuecomment-172387515>
> .
>
-- 

Sent from Mobile.

**Comment by mojavelinux** [comment]
Super! Let me know when it's ready and I'll do another pass at it. :memo:

**Comment by jaredmorgs** [comment]
Git is great when it works as you expected, but when it doesn't with regard to git --amend, you get the chook's breakfast you see above.  

In summary:

Resolves Issue #383 by further adding to the existing sectnumlevels attribute docs. Removed override info, added detail about Flexible Attribute to sectnums (it's worth noting), and declared the `setnumlevels` attribute as 'intrinsic' because based on the current guidance about attributes in  typeshttp://asciidoctor.org/docs/user-manual/#attributes, it appears as though `sectnumlevels` falls into this category.

**Comment by mojavelinux** [comment]
Nice work!

> Git is great when it works as you expect, but when it doesn't with regard to git --amend

Revising commits is something that is tricky to learn, but once you do, you'll realize that just a few commands give you everything you need. The key thing to remember is that git is just about passing around changes. Every commit is just a delta. Using git --amend or rebase, you are just telling it how to organize those deltas into a different batch. If everything goes absolutely horribly wrong, you can just diff two versions to get the effective change, store that somewhere, reset the branch and apply the changes.

Just always remember that git is just sets of changes. And those changes can be redefined. At least, once I saw it that way, it wasn't a black box for me anymore.

I'll do a rebase merge on this one to clean the history a bit.

**Comment by mojavelinux** [comment]
@jaredmorgs take note of a few conventions I followed in the revised commit https://github.com/asciidoctor/asciidoctor.org/commit/4b054755bb683fc17075a72aa900c1827c85d48c

* how I refer to the issue in the commit subject (first line) (`resolves #xyz subject line`)
* how I used an unordered list (Markdown syntax) to summarize the changes

I find this works really well and renders well on GitHub. That's the style we want to document in the contribution guide and consistently follow.

**Comment by jaredmorgs** [comment]
Noting this for the contributors guide. 👍

On Tue, 19 Jan 2016, 16:48 Dan Allen <notifications@github.com> wrote:

> Closed #477 <https://github.com/asciidoctor/asciidoctor.org/pull/477>.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/asciidoctor/asciidoctor.org/pull/477#event-518720464>.
>
-- 

Sent from Mobile.

**Pull request closed without merging**